kernel/kernel-api/src/main/java/io/delta/kernel/PaginatedScan.java

```diff
@@ -0,0 +1,66 @@
+package io.delta.kernel;
+
+import io.delta.kernel.data.FilteredColumnarBatch;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.utils.CloseableIterator;
+import java.util.Optional;
+
+/**
+ * Extension of {@link Scan} that supports pagination.
+ *
+ * <p>This interface allows consumers to retrieve scan results in discrete, ordered pages rather
+ * than all at once. This is particularly useful for large datasets where materializing the full
+ * result set would be expensive in terms of memory or compute resources.
+ *
+ * <p>Pagination is achieved via a combination of {@code pageSize} and {@code pageToken}. The {@code
+ * pageSize} controls how many Parquet data files are returned in each page, while the {@code
+ * pageToken} encodes the location of next batch to read and is used to resume the scan from exactly
+ * where the last page ended. For the first page, the {@code pageToken} should be {@code
+ * Optional.empty()} or empty.
+ *
+ * <p>Consumers typically use {@link PaginatedScan} in a loop: they call {@code getScanFiles()} to
+ * retrieve an iterator over the current page's scan files, After consuming the iterator, then call
```

Comment:
`files, After consuming the iterator, then call` --> can you please fix the wording/grammar here a bit?